src/rdkafka_transport.c

```diff
@@ -498,6 +500,14 @@ static int rd_kafka_transport_ssl_connect (rd_kafka_broker_t *rkb,
 	if (!SSL_set_fd(rktrans->rktrans_ssl, rktrans->rktrans_s))
 		goto fail;
 
+	mtx_lock(&rkb->rkb_lock);
+	rd_snprintf(name, sizeof(name), "%s", rkb->rkb_nodename);
+	mtx_unlock(&rkb->rkb_lock);
+	if ((t = strrchr(name, ':')))
+		*t = '\0';
+	if (!SSL_set_tlsext_host_name(rktrans->rktrans_ssl, name))
```

Comment:
I don't think this is worth using an option. For example, the official Java client is enabling this automatically and this doesn't seem configurable. I suppose this is done automatically by some layer abstraction. Except for `s_client`, I never found a client where this option was configurable.

As for IP, [RFC](https://tools.ietf.org/html/rfc3546#section-3.1) says:

>  Literal IPv4 and IPv6 addresses are not permitted in "HostName".

I think your suggestion is enough. If IPv6, I suppose we'll get brackets (since it's extracted from host:port notation)?